espnet2/fileio/vad_scp.py

```diff
@@ -0,0 +1,109 @@
+import collections.abc
+from pathlib import Path
+from typing import List, Union
+
+import numpy as np
+from typeguard import check_argument_types
+
+from espnet2.fileio.read_text import read_2column_text
+
+
+class VADScpReader(collections.abc.Mapping):
+    """Reader class for 'vad.scp'.
```

Comment:
It looks similar to `segment`
Can you clarify what is the different from `segment` and why/where it is used here or other appropriate places?

Reply:
The major difference between the `vad.scp` and `segments` is that `vad.scp` would focus on utterance level, while the `segments` are expected to focus on a whole session. In practice, both `segments` and `vad.scp` could be used as the input to format_wav_scp.py, where
- `segments` will be used to split the wav into utterances
- `vad.scp` will be used to trim the silence within the utterance (even within a sentence). A practical reason to use this is that we found segments file will become very large if it also considers the silence within utterances, so we use the `vad.scp` as a compact version of showing that.

This silence trim step with vad is an important step for unsupervised ASR training, so we 
- add a specific stage in the `uasr.sh` to conduct vad there
- initial the vadscp to standardize the procedure

Reply:
I will add notes for future reference.